src/common/snippets/src/lowered/pass/insert_specific_iterations.cpp

```diff
@@ -33,118 +87,112 @@ LinearIR::constExprIt InsertSpecificIterations::insert_copy_loop(LinearIR& linea
+    const auto new_id = loop_manager->replace_with_new_loop(linear_ir, begin, std::next(end), decomposed_loop_info, unified_loop_id);
+    decomposed_loop_end->set_id(new_id);
+    decomposed_loop_end->set_increment(decomposed_loop_info->get_increment());
+    if (const auto static_loop_end = ov::as_type_ptr<op::LoopEndStatic>(decomposed_loop_end)) {
+        static_loop_end->set_work_amount(decomposed_loop_info->get_work_amount());
+        static_loop_end->set_ptr_increments(decomposed_loop_info->get_ptr_increments());
+        static_loop_end->set_finalization_offsets(decomposed_loop_info->get_finalization_offsets());
     }
+    // Note: handlers must be run on the range started with the first operation in the loop body.
+    const auto handlers = decomposed_loop_info->get_handlers_by_type();
+    handlers.run(linear_ir, std::next(begin), end);
+}
 
-    const auto new_id = loop_manager->replace_with_new_loop(linear_ir, new_loop_begin_pos, new_loop_end_pos,
-                                                            original_loop_info->get_work_amount(), original_loop_info->get_increment(),
-                                                            new_entry_points, new_exit_points, loop_id);
-    const auto loop_end = ov::as_type_ptr<op::LoopEndStatic>(std::prev(new_loop_end_pos)->get()->get_node());
-    OPENVINO_ASSERT(loop_end, "Cloned Loop does not contain LoopEnd op at the expected place.");
-    loop_end->set_id(new_id);
-    return new_loop_begin_pos;
+bool InsertSpecificIterations::decompose(LinearIR& linear_ir, LinearIR::constExprIt begin, LinearIR::constExprIt end,
+                                         const std::shared_ptr<op::LoopEnd>& loop_end) {
+    const auto loop_id = loop_end->get_id();
+    const auto& loop_manager = linear_ir.get_loop_manager();
+    const auto& unified_loop_info = loop_manager->get_loop_info<UnifiedLoopInfo>(loop_id);
+
+    auto remaining_work_amount = unified_loop_info->get_work_amount();
+    const auto is_wa_dynamic = utils::is_dynamic_value(remaining_work_amount);
+
+    static constexpr std::array<SpecificLoopIterType, 3> loop_iterations = {
+        SpecificLoopIterType::FIRST_ITER, SpecificLoopIterType::MAIN_BODY, SpecificLoopIterType::LAST_ITER
+    };
+
+    auto decomposed = false;
+    for (const auto& iter_type : loop_iterations) {
+        if (is_decomposed_loop_needed(unified_loop_info, iter_type, remaining_work_amount)) {
+            const auto work_amount = get_decomposed_loop_work_amount(unified_loop_info, iter_type, remaining_work_amount);
+            const auto increment = get_decomposed_loop_increment(unified_loop_info, iter_type, remaining_work_amount);
+            // Update remaining Loop work amount
+            // Note: if work_amount is unknown and increment = 1, it means that a loop will iterate by whole work_amount
+            if ((!is_wa_dynamic) || (increment == 1 && utils::is_dynamic_value(work_amount))) {
```

Comment:
Maybe I'm missing smth, but do we really need the second condition?
I mean if `!is_wa_dynamic` is `false` then `utils::is_dynamic_value(work_amount)` is always `true`. It means that if we are checking the second condition, then work_amount is 100% dynamic, and we need to check only the increment. 
What do you think?
```suggestion
            if (!is_wa_dynamic || increment == 1) {
```